pkg/txm/metrics_test.go

```diff
@@ -71,19 +52,12 @@ func TestEmitTxMessage(t *testing.T) {
 		require.NoError(t, err)
 
 		// THEN
-		beholderEmitter.AssertCalled(t,
-			"Emit",
-			ctx,
-			mock.MatchedBy(func(b []byte) bool {
-				if err := proto.Unmarshal(b, &actualMessage); err != nil {
-					return false
-				}
-				return true
-			}),
-			"beholder_domain", "svr",
-			"beholder_entity", "svr.v1.TxMessage",
-			"beholder_data_schema", "/beholder-tx-message/versions/2",
-		)
+		messages := beholderTester.Messages(t)
+
+		assert.Len(t, messages, 1)
```

Comment:
Given that the two tests are run in parallel and `beholderTests.Beholder(t)` returns the same global object, I think it would be safer to match the message here exactly using `beholderTester.Len` or some other method I'm not aware of to avoid flakes.